tests/Services/AdSlotServiceTest.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Tests\Services;
+
+use Tests\Fixtures;
+use LiveIntent\AdSlot;
+use LiveIntent\Exceptions\InvalidRequestException;
+
+class AdSlotServiceTest extends ServiceTestCase
+{
+    protected $serviceKey = 'adSlots';
+
+    public function testIsFindable()
+    {
+        $adSlot = $this->service->find(Fixtures::adSlotId());
+        $this->assertInstanceOf(AdSlot::class, $adSlot);
+
+        $adSlot = $this->service->find(Fixtures::adSlotHash());
+        $this->assertInstanceOf(AdSlot::class, $adSlot);
+    }
+
+    public function testIsCreatableViaAttributesArray()
+    {
+        $adSlot = $this->service->create([
+            'name' => 'SDK Test',
+            'newsletter' => Fixtures::newsletterHash(),
+            'type' => 'image',
+            'mediaType' => 'newsletter',
+            'sizes' => [
+                [
+                    'width' => 500,
+                    'height' => 600,
+                    'floor' => 1.0,
+                    'deviceTypes' => [1,2,3],
+                ],
+            ],
+            'adIndicatorId' => 1,
+        ]);
+
+        $this->assertNotNull($adSlot->id);
+        $this->assertInstanceOf(AdSlot::class, $adSlot);
+    }
+
+    public function testIsCreatableViaResourceInstance()
+    {
+        $adSlot = new AdSlot([
+            'name' => 'SDK Test',
+            'newsletter' => Fixtures::newsletterHash(),
+            'type' => 'image',
+            'mediaType' => 'newsletter',
+            'sizes' => [
+                [
+                    'width' => 500,
+                    'height' => 600,
+                    'floor' => 1.0,
+                    'deviceTypes' => [1,2,3],
+                ],
+            ],
+            'adIndicatorId' => 1,
+        ]);
+
+        $adSlot = $this->service->create($adSlot);
+
+        $this->assertNotNull($adSlot->id);
+        $this->assertInstanceOf(AdSlot::class, $adSlot);
+    }
+
+    public function testIsUpdateableViaAttributesArray()
```

Comment:
Request:
Can we add a test to update ad slot sizes as well since that's what gamma-api will be doing.
Maybe there are two use cases here?
1. valid sizes (plural) update
2. invalid sizes update